src/base/RunnersRegistry.h

```diff
@@ -121,11 +121,34 @@ class IndependentRunner: public RegisteredRunner
 /// convenience function to "use" an otherwise unreferenced static variable
 bool UseThisStatic(const void *);
 
+// TODO: Remove, together with UseThisStatic(), after migrating to DefineRunnerRegistrator()
```

Comment:
This is a draft PR because it lacks conversion of the remaining dozen of RunnerRegistrationEntry() calls and the removal of the corresponding macros. I am posting this draft PR to solicit feedback about the proposed registration interface (so that we do not need to modify those dozens of calls multiple times).

To minimize the number of uncaught registration typos, I also would like to reject duplicate registrations (at startup time), but I need to double check that the corresponding (trivial) RegisterRunner_() enhancement will not break IndependentRunners first. We can now also add "registering such-and-such" debugging so it is easy to tell whether Squid recognized that a given module must be registered. These enhancements do not have to be done in this PR though.

Reply:
> This is a draft PR because it lacks conversion of the remaining dozen of RunnerRegistrationEntry() calls and the removal of the corresponding macros.

Done (as of branch commit c9edc47). This is no longer a draft PR.

> TODO: Reject duplicate registrations (at Squid startup time).

> To minimize the number of uncaught registration typos, I also would like to reject duplicate registrations (at startup time), but I need to double check that the corresponding (trivial) RegisterRunner_() enhancement will not break IndependentRunners first.

After examining the related code, I dropped the above TODO. I believe the enhancement mentioned above should be done in a dedicated PR because it requires a few code changes that have nothing to do with this PR. Technically, the official code also allows duplicate registrations, so this PR does not add that problem, and does not have to solve it.